# Data Analytics Internship Project 1

# 1. Import Libraries

In [1]:
import pandas as pd
import numpy as np

# 2. Load Dataset

In [2]:
df = pd.read_excel("Dataset for Data Analytics.xlsx")

df.head()

,OrderID,Date,CustomerID,Product,Quantity,UnitPrice,ShippingAddress,PaymentMethod,OrderStatus,TrackingNumber,ItemsInCart,CouponCode,ReferralSource,TotalPrice
0,ORD200000,2023-01-04,C72649,Monitor,5,570.62,928 Main St,Debit Card,Shipped,TRK37947903,7,SAVE10,Instagram,2853.10
1,ORD200001,2024-08-23,C75739,Phone,2,151.35,823 Main St,Online,Shipped,TRK91186779,3,SAVE10,Referral,302.70
2,ORD200002,2024-02-27,C81728,Tablet,5,550.68,512 Main St,Credit Card,Cancelled,TRK42903982,8,FREESHIP,Email,2753.40
3,ORD200003,2023-10-15,C33540,Chair,1,273.19,275 Main St,Debit Card,Returned,TRK62788070,5,SAVE10,Facebook,273.19
4,ORD200004,2025-05-08,C81840,Printer,4,626.01,668 Main St,Online,Delivered,TRK29241424,8,SAVE10,Email,2504.04


# 3. Understand the Dataset

### Check Shape

In [3]:
print("Rows and Columns:", df.shape)

Rows and Columns: (1200, 14)


### Check Column Names

In [4]:
print(df.columns)

Index(['OrderID', 'Date', 'CustomerID', 'Product', 'Quantity', 'UnitPrice',
       'ShippingAddress', 'PaymentMethod', 'OrderStatus', 'TrackingNumber',
       'ItemsInCart', 'CouponCode', 'ReferralSource', 'TotalPrice'],
      dtype='object')


### Check Data Types

In [5]:
print(df.dtypes)

OrderID                    object
Date               datetime64[ns]
CustomerID                 object
Product                    object
Quantity                    int64
UnitPrice                 float64
ShippingAddress            object
PaymentMethod              object
OrderStatus                object
TrackingNumber             object
ItemsInCart                 int64
CouponCode                 object
ReferralSource             object
TotalPrice                float64
dtype: object


### Dataset Information

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1200 entries, 0 to 1199
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype         
---  ------           --------------  -----         
 0   OrderID          1200 non-null   object        
 1   Date             1200 non-null   datetime64[ns]
 2   CustomerID       1200 non-null   object        
 3   Product          1200 non-null   object        
 4   Quantity         1200 non-null   int64         
 5   UnitPrice        1200 non-null   float64       
 6   ShippingAddress  1200 non-null   object        
 7   PaymentMethod    1200 non-null   object        
 8   OrderStatus      1200 non-null   object        
 9   TrackingNumber   1200 non-null   object        
 10  ItemsInCart      1200 non-null   int64         
 11  CouponCode       891 non-null    object        
 12  ReferralSource   1200 non-null   object        
 13  TotalPrice       1200 non-null   float64       
dtypes: datetime64[ns](1), float64(2), int64(

# 4. Statistical Summary

In [7]:
df.describe()

,Date,Quantity,UnitPrice,ItemsInCart,TotalPrice
count,1200,1200.000000,1200.000000,1200.000000,1200.000000
mean,2024-03-22 16:58:48,2.945833,356.412750,5.485000,1053.968300
min,2023-01-01 00:00:00,1.000000,11.390000,1.000000,11.390000
25%,2023-08-03 18:00:00,2.000000,186.062500,4.000000,410.520000
50%,2024-03-23 00:00:00,3.000000,364.210000,5.000000,823.615000
75%,2024-11-08 12:00:00,4.000000,521.570000,7.000000,1578.475000
max,2025-06-30 00:00:00,5.000000,699.930000,10.000000,3456.400000
std,NaN,1.407557,197.177146,2.281983,819.856558


# 5. Check Missing Values

### Count Missing Values

In [8]:
df.isnull().sum()

OrderID              0
Date                 0
CustomerID           0
Product              0
Quantity             0
UnitPrice            0
ShippingAddress      0
PaymentMethod        0
OrderStatus          0
TrackingNumber       0
ItemsInCart          0
CouponCode         309
ReferralSource       0
TotalPrice           0
dtype: int64

### Percentage of Missing Values

In [9]:
(df.isnull().sum()/len(df))*100

OrderID             0.00
Date                0.00
CustomerID          0.00
Product             0.00
Quantity            0.00
UnitPrice           0.00
ShippingAddress     0.00
PaymentMethod       0.00
OrderStatus         0.00
TrackingNumber      0.00
ItemsInCart         0.00
CouponCode         25.75
ReferralSource      0.00
TotalPrice          0.00
dtype: float64

### Observation

CouponCode = 309 missing values

# 6. Handle Missing Values

Since CouponCode is a categorical column:

In [12]:
df["CouponCode"] = df["CouponCode"].fillna("No Coupon")

#### Verify:

In [14]:
df.isnull().sum()

OrderID            0
Date               0
CustomerID         0
Product            0
Quantity           0
UnitPrice          0
ShippingAddress    0
PaymentMethod      0
OrderStatus        0
TrackingNumber     0
ItemsInCart        0
CouponCode         0
ReferralSource     0
TotalPrice         0
dtype: int64

# 7. Check Duplicate Records

### Entire Dataset Duplicates

In [18]:
df.duplicated().sum()

np.int64(0)

### Result:

0 Duplicate Rows

### Remove Duplicates

In [16]:
df = df.drop_duplicates()

# 8. Check Duplicate Order IDs

In [17]:
df["OrderID"].duplicated().sum()

np.int64(0)

### Result:

0 Duplicate Order IDs

# 9. Check Date Format

In [20]:
df["Date"]

0      2023-01-04
1      2024-08-23
2      2024-02-27
3      2023-10-15
4      2025-05-08
          ...    
1195   2024-06-20
1196   2024-03-04
1197   2023-07-13
1198   2024-08-22
1199   2023-06-11
Name: Date, Length: 1200, dtype: datetime64[ns]

### Convert to Proper Datetime Format

In [21]:
df["Date"] = pd.to_datetime(df["Date"])

### Verify

In [22]:
df["Date"].dtype

dtype('<M8[ns]')

### Result:

datetime64[ns]

No incorrect date formats found.

# 10. Check Numeric Columns

In [25]:
numeric_cols = df.select_dtypes(include=["int64","float64"]).columns

print(numeric_cols)

Index(['Quantity', 'UnitPrice', 'ItemsInCart', 'TotalPrice'], dtype='object')


# 11. Check Data Consistency

### Quantity Should Not Be Negative

In [26]:
(df["Quantity"] < 0).sum()

np.int64(0)

### Price Should Not Be Negative

In [27]:
(df["UnitPrice"] < 0).sum()

np.int64(0)

### Total Price Should Not Be Negative

In [28]:
(df["TotalPrice"] < 0).sum()

np.int64(0)

# 12. Check Unique Values

### Payment Methods

In [29]:
df["PaymentMethod"].unique()

array(['Debit Card', 'Online', 'Credit Card', 'Gift Card', 'Cash'],
      dtype=object)

### Order Status

In [30]:
df["OrderStatus"].unique()

array(['Shipped', 'Cancelled', 'Returned', 'Delivered', 'Pending'],
      dtype=object)

### Referral Source

In [31]:
df["ReferralSource"].unique()

array(['Instagram', 'Referral', 'Email', 'Facebook', 'Google'],
      dtype=object)

# 13. Check Outliers

### Quantity

In [32]:
Q1 = df["Quantity"].quantile(0.25)
Q3 = df["Quantity"].quantile(0.75)

IQR = Q3 - Q1

lower = Q1 - 1.5*IQR
upper = Q3 + 1.5*IQR

outliers = df[(df["Quantity"] < lower) | (df["Quantity"] > upper)]

print(outliers.shape)

(0, 14)


# 14. Create Clean Dataset

In [33]:
df.to_excel("Cleaned_Dataset.xlsx", index=False)

# 15. Final Validation

### Missing Values

In [34]:
df.isnull().sum()

OrderID            0
Date               0
CustomerID         0
Product            0
Quantity           0
UnitPrice          0
ShippingAddress    0
PaymentMethod      0
OrderStatus        0
TrackingNumber     0
ItemsInCart        0
CouponCode         0
ReferralSource     0
TotalPrice         0
dtype: int64

### Duplicate Rows

In [35]:
df.duplicated().sum()

np.int64(0)

### Duplicate IDs

In [36]:
df["OrderID"].duplicated().sum()

np.int64(0)

### Date Format

In [37]:
df["Date"].dtype

dtype('<M8[ns]')

# Conclusion

This project focused on the **Data Cleaning and Preparation** phase of the data analytics process. The dataset was thoroughly examined to identify missing values, duplicate records, and data formatting issues. Missing values in the **CouponCode** column were successfully handled, duplicate records and duplicate Order IDs were checked, and date formats were verified and standardized. Additionally, the dataset was inspected for data consistency and potential outliers to ensure accuracy and reliability.

After performing all cleaning operations, the dataset became complete, consistent, and ready for further analysis and visualization. This project demonstrated the importance of data quality in analytics, as accurate and well-prepared data leads to more reliable insights and better decision-making. Through this project, key skills in Python, Pandas, data cleaning, and data preparation were applied successfully, providing a strong foundation for future data analytics tasks.
